### PR TITLE
[ws-manager-mk2] Protect tokens

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Store sensitive workspace data (tokens) in short lived Kubernetes secrets. When a workspace start is requested we create a secret (named [workspaceid]-tokens) that contains the tokens extracted from the content initializer. The tokens will be removed from the content initializer. 

The secret itself will live in a separate namespace so that we can narrow down the scope of permissions we need to give to ws-daemon so that it can access the secrets. When the content init happens in ws-daemon the secrets will be injected into the initializer again and the secret will be deleted once the workspace reaches the `running` phase but latest when it reaches the `stopped` phase.

## Related Issue(s)
n.a.

## How to test
- Open workspace for a private repository

## Release Notes

```release-note
None
```
## Build Options:

- [x] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [] /werft with-preview
- [] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/ws-manager-mk2/controllers/workspace_controller.go [hunk 1]
```diff
@@ -358,6 +358,11 @@ func (r *WorkspaceReconciler) deleteWorkspaceSecrets(ctx context.Context, ws *wo
 	if err != nil {
 		log.Error(err, "could not delete environment secret", "workspace", ws.Name)
 	}
+
+	err = r.deleteSecret(ctx, fmt.Sprintf("%s-%s", ws.Name, "tokens"), r.Config.SecretsNamespace)
+	if err != nil {
+		log.Error(err, "could not delete token secret", "workspace", ws.Name)
+	}
```
Comment (WVerlaek) [inline components/ws-manager-mk2/controllers/workspace_controller.go:365]: still think we should return an error when we fail to delete the secret, and not remove the finalizer if there's an error, so we try again until all secrets have been cleaned up

**Review by WVerlaek — APPROVED** [review]
lgtm, tested in preview env. Added hold for 2 comments

File: .werft/jobs/build/installer/post-process.sh [hunk 2]
```diff
@@ -52,7 +52,7 @@ MATCHES="$(grep -c -- --- k8s.yaml)"
 # get the read number of K8s manifest docs
 # K8s object names and kinds are duplicated in a config map to faciliate deletion
 # subtract one (the config map) and then divide by 2 to get the actual # of docs we'll loop through
-DOCS="$((((MATCHES - 1) / 2) + 1))"
```
[the remark this answers is not in the document]
Comment (Furisto) [reply .werft/jobs/build/installer/post-process.sh:55]: Had an out of bounds error without it.

File: components/ws-manager-mk2/controllers/workspace_controller.go (hunk 1 above)
Comment (Furisto) [reply components/ws-manager-mk2/controllers/workspace_controller.go]: We retry the deletion of the secret now and will not delete the workspace CR until the deletion of the secret suceeds.

**Pull request merged**